### PR TITLE
POC loads on PointOfCareDashboard within POC app and supports selection

### PR DESCRIPTION
**Pull request by karenzheung**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

## Submitter:

JIRA 1960

Go to endpoint POC to see the content. Programmatically added a clinical note with a placeholder in order for the POC to render. 

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [x] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [x] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [N/A] Document any new APIs/extension points
- [N/A ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [X] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [X] Code is commented

**Tests**

- [X] Existing tests passed
- [X] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by sgober** [comment]
This is going to have a conflicting rebase with my PR.  I suggest we try to get mine in and maybe copy and paste the changes? might be easier...

**Comment by Dtphelan1** [comment]
This can now be rebased/manually combined with Sarah's most recent changes!

**Comment by sgober** [comment]
@karenzheung I'm going to hold off on this until the changes are combined with mine.  Might be easiest to copy/paste them instead of rebase!

**Comment by karenzheung** [comment]
This has been rebased

File: src/panels/PointOfCarePanel.jsx [hunk 1]
```diff
@@ -1,7 +1,37 @@
...
+        let noteContent = '@condition[[{"text":"Gastrointestinal stromal tumor","entryId":"8"}]] <disease status>';
+        const entryId = patient.addClinicalNote('', '', '', '', '', noteContent, false);
+        const updatedEditorNote = patient.getEntryById(entryId);
+        const placeholder = shortcutManager.createPlaceholder('#disease status', '<disease status>', undefined, contextManager, patient, updatedEditorNote, false);
+        structuredFieldMapManager.addPlaceholder(placeholder);
+
+        return (
+            <div id="point-of-care-dashboard-content" style={{display: "flex"}}>
+                <PointOfCare
+                structuredFieldMapManager={structuredFieldMapManager}
```
Comment (Dtphelan1) [inline src/panels/PointOfCarePanel.jsx]: Could we get an indent here?

File: src/dashboard/PointOfCareDashboard.jsx [hunk 2]
```diff
@@ -143,7 +143,9 @@ export default class PointOfCareDashboard extends Component {
 
         return (
             <div className="right-border-box point-of-care-container" style={PointOfCarePanelStyles}>
-                <PointOfCarePanel />
+                <PointOfCarePanel
+                appState={this.props.appState}
```
Comment (Dtphelan1) [inline src/dashboard/PointOfCareDashboard.jsx]: Could we get an indent here too?

**Comment by Dtphelan1** [comment]
Ahh I just added the indents myself. Overall I think this looks good! Checking one or two more things on my end, but approval should be incoming!

**Comment by Dtphelan1** [comment]
Spoke too soon - [DP1] Can't actually select values in the POC. When I try, I get the following error. 
image

Should be fixed if we just make sure our note/contextManager have a valid parent context in scope.

**Comment by sgober** [comment]
Rebased and updated some linting warnings.  Can confirm that I am also seeing the error @Dtphelan1 is getting.

**Comment by sgober** [comment]
Handled travis failing... had some linting warnings.  Also rebased.

File: src/containers/FullApp.jsx [hunk 3]
```diff
@@ -278,6 +278,8 @@ export class FullApp extends Component {
     newCurrentShortcut = (shortcutC, shortcutType, shortcutData, updatePatient = true, source) => {
 
         let newShortcut = this.shortcutManager.createShortcut(shortcutC, shortcutType, this.state.patient, shortcutData, this.handleShortcutUpdate);
+        console.log(newShortcut);
```
Comment (sgober) [inline src/containers/FullApp.jsx]: Can we remove these log statements?

File: src/dashboard/PointOfCareDashboard.jsx [hunk 4]
```diff
@@ -141,9 +191,20 @@ export default class PointOfCareDashboard extends Component {
             "transition": "width .5s",
         };
 
+        const note = this.getFirstInProgressNote();
+        console.log(note);
```
Comment (sgober) [inline src/dashboard/PointOfCareDashboard.jsx]: and these log statements too

File: src/notes/FillPlaceholder.jsx [hunk 5]
```diff
@@ -154,9 +154,12 @@ export default class FillPlaceholder extends Component {
     };
 
     onSetValue = (source, attributeSpec, entryIndex, newValue, moveToNextField = true) => {
+        console.log("here");
         const { placeholder } = this.props;
         //if (entryIndex === -1) entryIndex = placeholder.entryShortcuts.length - 1;
         const attributes = placeholder.getAttributeValue(attributeSpec.name, entryIndex);
+        console.log('source ', source);
+        console.log('placeholder ', placeholder);
```
Comment (sgober) [inline src/notes/FillPlaceholder.jsx]: and these

File: src/notes/FluxNotesEditor.jsx [hunk 6]
```diff
@@ -284,6 +284,7 @@ class FluxNotesEditor extends React.Component {
 
     newPlaceholder = (placeholderText, data) => {
         const shortcutName = "#" + placeholderText.substring(1, placeholderText.length-1); // strip off < and > and add #
+        console.log(this.props.selectedNote);
```
Comment (sgober) [inline src/notes/FluxNotesEditor.jsx]: and these

File: src/panels/PointOfCarePanel.jsx [hunk 7]
```diff
@@ -1,7 +1,96 @@
...
+import ShortcutManager from '../shortcuts/ShortcutManager';
 
 export default class PointOfCarePanel extends Component {
+
+    setUpFakeClinicialNote() {
+        this.structuredFieldMapManager = new StructuredFieldMapManager();
+        this.shortcutManager = new ShortcutManager();
+        const patient = this.props.appState.patient;
+        this.contextManager = this.props.contextManager;
+        console.log('contextManager ', this.contextManager);
```
Comment (sgober) [inline src/panels/PointOfCarePanel.jsx]: here too

File: src/shortcuts/Placeholder.jsx [hunk 8]
```diff
@@ -105,6 +105,7 @@ class Placeholder {
     }
 
     getTextWithStylingToDisplayInNote(index = 0) {
+        console.log(this);
```
Comment (sgober) [inline src/shortcuts/Placeholder.jsx]: last one

**Comment by sgober** [comment]
[SO1] there are a lot of log statements that need to be removed

**Comment by sgober** [comment]
I'm seeing this work great now, nice job @gregquinn2001 and @karenzheung!

[SO1] There are still some remaining log statements that need to be taken out.

I fixed the remaining linting errors and rebased, hopefully that should get rid of travis failing.

**Comment by sgober** [comment]
Never mind...

[SO3] two tests are failing

**Comment by gregquinn2001** [comment]
@sgober  I had resolved all linter errors and removed all console logs so I'm very confused by what you did? I'm having all sorts of conflicts when I try to pull the latest branch now as well. @karenzheung do you see any issues with the branch?

File: src/model/oncocore/FluxCancerProgression.js [hunk 9]
```diff
@@ -53,6 +53,7 @@ export default class FluxCancerProgression extends FluxEntry {
      *  The method will lookup the corresponding coding/codesystem and set the _codeableConcept property
      */
     set status(status) {
+        console.log("************************************** FluxCancerProgression.status set to " + status);
```
Comment (sgober) [inline src/model/oncocore/FluxCancerProgression.js]: remove console log

**Comment by sgober** [comment]
@gregquinn2001 I just commented where the log is. For the linting, I was trying to figure out why travis was failing so I ran `yarn lint` and there were still errors, so that's what I fixed.  However there still are two tests that are failing.

You're probably having a hard time pulling because I rebased with master.

**Comment by karenzheung** [comment]
I see the two test that are failing, but pulling isn't giving me problems

**Comment by gregquinn2001** [comment]
@sgober and @karenzheung  let's wait for my pull request to get merged. I should be involved in the rebase because Karen cherry picked some code from my pull request that code subsequently changed on my branch. It will be more efficient if I'm involved. The tests should pass after that.

**Comment by gregquinn2001** [comment]
Removed console log so SO1 is addressed. thanks!

**Comment by gregquinn2001** [comment]
@karenzheung the rebase to my branch should fix one of the tests. The other test (FullApp.test) is broken due to the commented out first line of code in ContextManager.addShortcutToContext. I'm going to look into a fix.

**Comment by gregquinn2001** [comment]
Fixed one of the failing tests for SO3 (FullApp.test). Of course SO2 is resolved as well. Still have the failing NoteParser test though so SO3 not fully resolved.

**Comment by gregquinn2001** [comment]
I pulled in fix from other branch. SO3 is resolved.

**Comment by gregquinn2001** [comment]
Did rebase after my branch was merged to master. actually was not too bad.

**Comment by Dtphelan1** [comment]
This all looks good to me now. This task accomplishes what it set out to, and I don't have any additional comments on how we got that done. I think maybe longterm we should have some discussions on the necessity of note-creation when capturing structured data in our app, but that's well beyond the scope of this PR. Great work @karenzheung  and @gregquinn2001!  I'm going to try and rebase with master and if that's good I'll merge.

**Comment by Dtphelan1** [comment]
Tests pass - merging

**Pull request merged**